src/main/java/seedu/duke/validation/Validation.java

```diff
@@ -0,0 +1,203 @@
+        if (taskString.equals("")) {
+            if (task.equals("delete") || task.equals("done")) {
+                throw new DukeException("The index for a " + task + " command cannot be empty.");
+            } else if (task.charAt(0) == 'a' || task.charAt(0) == 'e' || task.charAt(0) == 'i'
+                    || task.charAt(0) == 'o' || task.charAt(0) == 'u') {
+                throw new DukeException("The description of an " + task + " cannot be empty.");
+            } else {
+                throw new DukeException("The description of a " + task + " cannot be empty.");
+            }
+        }
+    }
+
+    /**
+     * Ensures that a user does not enter an empty line of command.
+     * @param command String representation of a single-word command to the program.
+     * @throws DukeException If user inputs an empty line of command.
+     */
+    public static void ensureNonEmptyCommand(String command) throws DukeException {
+        if (command.equals("")) {
+            throw new DukeException("Command cannot be empty.");
+        }
+    }
+
+    /**
+     * Returns a validated Date object.
+     * Ensures that input String is in valid DD/MM/YYYY format.
+     * Ensures there are at most 12 months in a year.
+     * Ensures the day of the month in the year entered is valid.
+     * @param date String representation of a date in DD/MM/YYYY format.
+     * @return Validated Date object containing the day, month, year of the date parameter.
+     * @throws DukeException If the date entered is invalid.
+     */
+    public static Date getValidatedDate(String date) throws DukeException {
+        if (date.length() != 10) {
+            throw new DukeException("Please ensure date format is in DD/MM/YYYY");
+        }
+
+        String[] dateArr = date.split("/");
+        if (dateArr.length != 3) {
+            throw new DukeException("Please ensure date format is in DD/MM/YYYY");
+        }
+
+        int day = 0;
+        int month = 0;
+        int year = 0;
+        boolean isLeapYear = false;
+
+        try {
+            day = Integer.parseInt(dateArr[0]);
+            month = Integer.parseInt(dateArr[1]);
+            year = Integer.parseInt(dateArr[2]);
+            if (day < 1) {
+                throw new DukeException("Calendar day must start from 1");
+            } else if (month < 1) {
+                throw new DukeException("Calendar month must start from 1");
+            } else if (year < 0) {
+                throw new DukeException("Calendar year must start from 0");
+            } else if (month > 12) {
+                throw new DukeException("There cannot be more than 12 months in a year.");
+            }
+        } catch (NumberFormatException ex) {
+            throw new DukeException("Please ensure date format is in DD/MM/YYYY");
+        }
+
+        if (year % 4 == 0) {
+            isLeapYear = true;
+        } else if (year % 100 == 0 && year % 400 == 0) {
+            isLeapYear = true;
+        }
```

Comment:
It seems that a year which is divisible by 100 but not divisible by 400 are also determined as leap year here.

Reply:
Thanks for helping to spot the logic bug!

Have corrected the code to behave as follows instead:

```suggestion
        if (year % 4 == 0 && year % 100 != 0) {
            isLeapYear = true;
        } else if (year % 400 == 0) {
            isLeapYear = true;
        }
```